### ISSUE TITLE
trailing-comma and rest args.... it is broken

### ISSUE THREAD

**Issue by brikou** [palantir/tslint#3683]
### Bug Report

- __TSLint version__: latest
- __TypeScript version__: latest
- __Running TSLint via__: (pick one) CLI / Node.js API / VSCode / grunt-tslint / Atom / Visual Studio / etc

#### TypeScript code being linted

```ts
// code snippet
const Foo = ({
    bar,
    ...props // never ever add a comma otherwise, rest won't work /!\
}) => <h1>Foo</h1>
```

with `tslint.json` configuration:

```json
default from init recommended
```

#### Actual behavior

Tslint add a comma after rest "...props"

#### Expected behavior

Never add a comma after rest "...props"

**Comment by ajafff** [comment]
> never ever add a comma otherwise, rest won't work /!\

This is totally valid TypeScript code, hence the default config enforces trailing commas.

Please read [**the docs**](https://palantir.github.io/tslint/rules/trailing-comma/). There's an option to disable this behavior.

> To align this rule with the ECMAScript specification that is implemented in modern JavaScript VMs, there is a third option esSpecCompliant. Set this option to true to disallow trailing comma on object and array rest and rest parameters.

**Comment by brikou** [comment]
Sorry if I miss some details and didn't took to explain further. I like the default comma rule but when a rest param is last parameter,always rule should have an exception. Otherwise code is broken. Prettier handle this edgecase as expected, and tslint should IMO not produce broken code.

**Comment by ajafff** [comment]
I totally get your point. This has been the default since this rule was introduced. I added the `esSpecCompliant` a few versions ago. Changing the default now would be a breaking change, because users get an error about the forbidden comma where the linter previously enforced one.
Since the primary target of TSLint is code that gets transpiled by the TypeScript compiler, this is not a big problem, because TypeScript allow the trailing comma in this position and removes it during compilation.

If you don't want this default behavior, you need to override the rule's options.

**Comment by brikou** [comment]
No it didn't removed it during compilation, I was using target esnext (which probably prevent ts from removing it)
